metrics/test/test_recursion.py

```diff
@@ -25,3 +25,23 @@ def test_recursion_given_recursion_file_returns_1(self):
         actual = len(self._metric.run())
         expected = 1
         self.assertEqual(actual, expected)
+
+    def test_recursion_given_c_sharp_file_returns_1(self):
+        new_source_repository = SourceRepository(
+            Path("./metrics/test/data/test_data_recursion"), "c_sharp"
+        )
```

Comment:
Couldn't you just set the language field of the current source_repository?

```py
self._source_repository.language = "c_sharp"
```

Reply:
My guess is, its because source_repository only determine what files to inspect when it is constructed. Lets add a notion task for it.